tiledb/sm/query/query_buffer.h

```diff
@@ -147,6 +147,32 @@ class QueryBuffer {
     return *this;
   }
 
+  /** Move-assignment Operator. */
+  Status set_buffer_data(QueryBuffer&& rhs) {
```

Comment:
To clarify on my comment from the original review: let's have a setter for each type:
```
Status set_buffer_data(void* data_buffer, uint64_t size);
Status set_buffer_data_var(void* data_var_buffer, uint64_t size);
Status set_buffer_offsets(void* offsets_buffer, uint64_t size);
Status set_buffer_validity(ValidityVector&& validity_vector);
```

If the buffer being set is non-null, return a non-OK status.

Reply:
If the buffer is _non_-null? Isn't this the case in which we set the params using `std::swap()`?